Include/cpython/pyhash.h

```diff
@@ -17,6 +17,7 @@
 
 #define _PyHASH_MODULUS (((size_t)1 << _PyHASH_BITS) - 1)
 #define _PyHASH_INF 314159
+#define _PyHASH_NAN 0
```

Comment:
IMO, it'd be better to remove this: the documentation is clear enough with just ``0`` and I expect the same for the code.

Reply:
Once Py_HashDouble() will be added, I will consider exposing most PyHASH constants, it was requested by: https://github.com/python/cpython/issues/111389

Reply:
Hum, in fact, I changed the PR to remove _PyHASH_NAN constant.